modal/functions.py

```diff
@@ -276,7 +332,158 @@ class _Function(_Object, type_prefix="fu"):
+                use_method_name=method_name,
+                existing_function_id=existing_object_id or "",
+            )
+            assert resolver.client.stub  # client should be connected at this point
+            response = await retry_transient_errors(resolver.client.stub.FunctionPrecreate, req)
+            method_bound_function._hydrate(response.function_id, resolver.client, response.handle_metadata)
+
+        def _deps():
+            return [class_function]
+
+        rep = f"Method({full_name})"
+
+        fun = _Function._from_loader(_load, rep, preload=_preload, deps=_deps)
+        fun._tag = full_name
+        fun._raw_f = partial_function.raw_f
+        fun._info = FunctionInfo(partial_function.raw_f, cls=user_cls, serialized=serialized)  # needed for .local()
+        fun._use_method_name = method_name
+        fun._app = class_function._app
+        fun._is_generator = partial_function.is_generator
+        fun._all_mounts = class_function._all_mounts
+        # TODO: set more attributes?
+
+        return fun
+
+    def _bind_instance_method(self, class_bound_method: "_Function"):
+        """mdmd:hidden
+
+        Binds an "instance service function" to a specific method.
+        This "dummy" _Function gets no unique object_id and isn't backend-backed at the moment, since all
+        it does it forward invocations to the underlying instance_bound_class_function with the specified method,
+        and we don't support web_config for parameterized methods at the moment.
+        """
+        instance_service_function = self
+        assert instance_service_function._obj
+        method_name = class_bound_method._use_method_name
+        full_function_name = f"{class_bound_method._function_name}[parameterized]"
+
+        def hydrate_from_instance_function(obj):
+            obj._hydrate_from_other(instance_service_function)
+            obj._obj = instance_service_function._obj
+            obj._web_url = class_bound_method._web_url  # TODO: this shouldn't be set when actual parameters are used
+            obj._function_name = full_function_name
+            obj._is_generator = class_bound_method._is_generator
+            obj._use_method_name = method_name
+            obj._use_function_id = instance_service_function.object_id
+
+        async def _load(fun: "_Function", resolver: Resolver, existing_object_id: Optional[str]):
+            # there is currently no actual loading logic executed to create each method on
+            # the *parameterized* instance of a class - it uses the parameter-bound service-function
+            # for the instance. This load method just makes sure to set all attributes after the
+            # `instance_service_function` has been loaded (it's in the `_deps`)
+            hydrate_from_instance_function(fun)
+
+        def _deps():
+            return [instance_service_function]
+
+        rep = f"Method({full_function_name})"
+
+        fun = _Function._from_loader(
```

Comment:
This uses a "loader" without having any actual rpc calls, just to ensure that the instance_service_function is loaded first as a dep before setting the hydration attributes on the "Pure Placeholder" function. Could clean up with some refactoring